### PR TITLE
feat(api): impl some web3 interfaces

### PR DESCRIPTION
**Pull request by DavidChild**
**Which docs this PR relation**:
1. sha3
2. accounts
3. clientVersion

**Which toolchain this PR adaption**:

No Breaking Change

**Special notes for your reviewer**:

### CODE REVIEW

File: devtools/chain/config.toml [hunk 1]
```diff
@@ -11,12 +11,12 @@ http_listening_address = "127.0.0.1:8000"
 ws_listening_address = "127.0.0.1:8010"
 maxconn = 25000
 max_payload_size = 1048576
+client_version = "Geth/v1.10.14-unstable-72c2c0ae/linux-amd64/go1.17.5"
```
Comment (driftluo) [inline devtools/chain/config.toml]: I don't think the version number has to be this value, I think a better way would be to get it like this https://github.com/nervosnetwork/ckb/blob/develop/build.rs#L31

File: core/api/src/jsonrpc/mod.rs [hunk 2]
```diff
@@ -75,6 +75,15 @@ pub trait AxonJsonRpc {
 
     #[method(name = "eth_getLogs")]
     async fn get_logs(&self, filter: Web3Filter) -> RpcResult<Vec<Web3Log>>;
+
+    #[method(name = "web3_clientVersion")]
+    async fn get_client_version(&self) -> RpcResult<String>;
```
Comment (KaoImin) [inline core/api/src/jsonrpc/mod.rs]: The `get` prefix is unnecessary.

File: core/api/src/jsonrpc/mod.rs [hunk 3]
```diff
@@ -75,6 +75,15 @@ pub trait AxonJsonRpc {
...
+
+    #[method(name = "eth_accounts")]
+    async fn get_accounts(&self) -> RpcResult<Vec<Hex>>;
```
Comment (KaoImin) [inline core/api/src/jsonrpc/mod.rs]: Move this upper.

File: core/api/src/jsonrpc/mod.rs [hunk 4]
```diff
@@ -75,6 +75,15 @@ pub trait AxonJsonRpc {
...
+
+    #[method(name = "web3_sha3")]
+    async fn get_sha3(&self, data: Hex) -> RpcResult<H256>;
```
Comment (KaoImin) [inline core/api/src/jsonrpc/mod.rs]: The `get` prefix is unnecessary.

File: devtools/chain/config.toml (hunk 1 above)
Comment (KaoImin) [reply devtools/chain/config.toml]: Why use geth version here?

**Review by KaoImin — CHANGES_REQUESTED** [review]

File: core/api/src/jsonrpc/impl.rs [hunk 5]
```diff
@@ -416,6 +420,21 @@ impl<Adapter: APIAdapter + 'static> AxonJsonRpcServer for JsonRpcImpl<Adapter> {
...
+
+    async fn get_client_version(&self) -> RpcResult<String> {
+        Ok(self.version.clone())
+    }
+
+    async fn get_accounts(&self) -> RpcResult<Vec<Hex>> {
+        Ok(vec![])
+    }
+
+    async fn get_sha3(&self, data: Hex) -> RpcResult<H256> {
```
Comment (KaoImin) [inline core/api/src/jsonrpc/impl.rs]:
```
    async fn get_sha3(&self, data: Hex) -> RpcResult<Hash> {
```

File: core/api/src/jsonrpc/mod.rs (hunk 4 above)
Comment (KaoImin) [inline core/api/src/jsonrpc/mod.rs]:
```
    async fn get_sha3(&self, data: Hex) -> RpcResult<Hash>;
```

File: devtools/chain/config.toml (hunk 1 above)
Comment (DavidChild) [reply devtools/chain/config.toml]: > Why use geth version here?

Original Ethereum dev chain `web3_clientVersion`  returns this value: `geth / v1 10.14-unstable-72c2c0ae/linux-amd64/go1. 17.5`, can I set empty value  in the configuration file or other value？
if return other value ,will it affect the user who calls this interface.

**Review by KaoImin — DISMISSED** [review]

File: core/api/src/jsonrpc/impl.rs [hunk 6]
```diff
@@ -471,7 +471,6 @@ impl<Adapter: APIAdapter + 'static> AxonJsonRpcServer for JsonRpcImpl<Adapter> {
         Ok(all_logs)
     }
 
-
```
Comment (KaoImin) [inline core/api/src/jsonrpc/impl.rs]: Why do this?

Comment (DavidChild) [reply core/api/src/jsonrpc/impl.rs]: > Why do this?
i not modify the code ,just format the code.

Comment (KaoImin) [reply core/api/src/jsonrpc/impl.rs]: Sorry, I thought blank lines were removed, but it wasn't.

**Review by KaoImin — APPROVED** [review]

**Comment by KaoImin** [comment]
LGTM

**Pull request merged**